scripts/npm-utils.js

```diff
@@ -21,14 +21,14 @@ const fs = require('fs'); // [macOS]
 
 // [macOS] Function to get our version from package.json instead of the CircleCI build tag.
 function getPkgJsonVersion() {
-  const RN_PACKAGE_DIRECTORY = path.resolve(
+  const RN_PACKAGE_DIRECTORY = path.join(
     __dirname,
     '..',
     'packages',
     'react-native',
   );
-  const pkgJsonPath = path.resolve(RN_PACKAGE_DIRECTORY, 'package.json');
-  const pkgJson = JSON.parse(fs.readFileSync(pkgJsonPath, 'utf8'));
+  const pkgJsonPath = path.join(RN_PACKAGE_DIRECTORY, 'package.json');
+  const pkgJson = JSON.parse(fs.readFileSync(pkgJsonPath, 'utf-8'));
```

Comment:
Prior to changing to `path.join`, I was seeing this error:

 ```
 ● publish-npm › release › should publish next

    TypeError: require(...).resolve is not a function

      22 | // [macOS] Function to get our version from package.json instead of the CircleCI build tag.
      23 | function getPkgJsonVersion() {
    > 24 |   const RN_PACKAGE_DIRECTORY = path.resolve(
         |                                     ^
      25 |     __dirname,
      26 |     '..',
      27 |     'packages',

      at resolve (scripts/npm-utils.js:24:37)
      at Object.getPkgJsonVersion [as getNpmInfo] (scripts/npm-utils.js:68:5)
      at getNpmInfo (scripts/publish-npm.js:66:26)
      at Object.<anonymous> (scripts/__tests__/publish-npm-test.js:226:7)
   ```